shell/browser/api/atom_api_url_request.cc

```diff
@@ -166,7 +173,9 @@ class ChunkedDataPipeGetter : public UploadDataPipeGetter,
   mojo::ReceiverSet<network::mojom::ChunkedDataPipeGetter> receiver_set_;
 };
 
-URLRequest::URLRequest(gin::Arguments* args) : weak_factory_(this) {
+URLRequest::URLRequest(gin::Arguments* args)
+    : id_(GetAllRequests().Add(this)), weak_factory_(this) {
```

Comment:
Can we maintain an IDMap of weak ptrs rather than raw ptrs , we already check before using it in `AuthResponder::OnAuthRequired`, this will just make it more stronger.

Reply:
no point in having a weak ptr, the object removes itself from the map on destruction so a weak ptr wouldn't add anything here.